workers/modules/shared-worker-parse-error-failure.html

```diff
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
+<title>SharedWorker: parse error failure</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+
+// Check if module shared worker is supported.
+// In this test scope, we only use simple non-nested static import as a feature
+// of module shared worker, so we only check if static import is supported.
+//
+// This check is necessary to appropriately test parse error handling because
+// we need to distingusih the parse error invoked by unsupported "import" in
+// the top-level script from the parse error invoked by the statically imported
+// script which is the one we want to check in this test.
+promise_setup(async () => {
+  const scriptURL = 'resources/static-import-worker.js';
+  const worker = new SharedWorker(scriptURL, { type: 'module' });
+  const supportsModuleWorkers = await new Promise((resolve, reject) => {
+    worker.port.onmessage = e => {
+      resolve(e.data.length == 1 && e.data[0] == 'export-on-load-script.js');
+    };
+    worker.onerror = () => reject(false);
+  });
+  assert_precondition(
+    supportsModuleWorkers,
+    "Static import must be supported on module shared worker to run this test."
+  );
+});
+
+const checkArguments = (args) => {
+  assert_equals(args.a.constructor, Event);
+  assert_true('message' in args.a, 'Event.message');
+  assert_equals(typeof args.a.message, 'string', 'Event.message');
+  assert_equals(args.a.filename, location.href + '/resources/syntax-error.js'),
+                'Event.filename');
+  assert_equals(args.a.lineno, 1, 'Event.lineno');
+  assert_equals(args.b, undefined, 'unexpected second argument to onerror');
+  assert_equals(args.c, undefined, 'unexpected third argument to onerror');
+};
+
+promise_test(async () => {
+  const scriptURL = 'resources/syntax-error.js';
+  const worker = new SharedWorker(scriptURL, { type: 'module' });
+  const args = await new Promise(resolve =>
+      worker.onerror = (a, b, c) => resolve({a, b, c}));
```

Comment:
This could be a little simpler as

```js
  const args = await new Promise(resolve => worker.onerror = (...args) => resolve(args));
```

and then `args` will be an array, which avoids the awkward a/b/c properties and lets the `checkArguments` function do something like `assert_equals(args.length, 1)`.